src/lib/uri/uri.h

```diff
@@ -154,6 +154,14 @@ uri_escape(const char *src, size_t src_size, char *dst,
 size_t
 uri_unescape(const char *src, size_t src_size, char *dst, bool decode_plus);
 
+/** Determine if uris refer to the same host:service or unix socket path. */
+bool
+uri_equals(const struct uri *a, const struct uri *b);
+
+/** Check if a uri isn't empty. */
+bool
+uri_is_set(const struct uri *uri);
```

Comment:
Why not `uri_is_nil`? That would be more like `tt_uuid_is_nil` which serves the same purpose.

Reply:
Makes sense, done:

```diff
diff --git a/src/lib/uri/uri.c b/src/lib/uri/uri.c
index f64b1055e..7a162c266 100644
--- a/src/lib/uri/uri.c
+++ b/src/lib/uri/uri.c
@@ -484,7 +484,7 @@ uri_equals(const struct uri *a, const struct uri *b)
 }
 
 bool
-uri_is_set(const struct uri *uri)
+uri_is_nil(const struct uri *uri)
 {
-       return memcmp(uri, &uri_nil, sizeof(*uri)) != 0;
+       return memcmp(uri, &uri_nil, sizeof(*uri)) == 0;
 }
diff --git a/src/lib/uri/uri.h b/src/lib/uri/uri.h
index 846fc4ec8..b068592b6 100644
--- a/src/lib/uri/uri.h
+++ b/src/lib/uri/uri.h
@@ -158,9 +158,9 @@ uri_unescape(const char *src, size_t src_size, char *dst, bool decode_plus);
 bool
 uri_equals(const struct uri *a, const struct uri *b);
 
-/** Check if a uri isn't empty. */
+/** Check if a uri is empty. */
 bool
-uri_is_set(const struct uri *uri);
+uri_is_nil(const struct uri *uri);
 
 #if defined(__cplusplus)
 } /* extern "C" */
```

and in the last commit:
```diff
diff --git a/src/box/replication.cc b/src/box/replication.cc
index 11af1d094..582a97f9b 100644
--- a/src/box/replication.cc
+++ b/src/box/replication.cc
@@ -964,7 +964,7 @@ is_applier_cfg_bootstrap_leader(const struct applier *applier)
                return tt_uuid_compare(&applier->uuid,
                                       &cfg_bootstrap_leader_uuid) == 0;
        }
-       if (uri_is_set(&cfg_bootstrap_leader_uri))
+       if (!uri_is_nil(&cfg_bootstrap_leader_uri))
                return uri_equals(&applier->uri, &cfg_bootstrap_leader_uri);
        return false;
 }
```